modules/adfs/lib/IdP/ADFS.php

```diff
@@ -38,6 +38,7 @@ public static function receiveAuthnRequest(SimpleSAML_IdP $idp) {
 			'ForceAuthn' => $forceAuthn,
 			'isPassive' => $isPassive,
 			'adfs:wctx' => $wctx,
+			'adfs:wreply' => (array_key_exists('wreply',$_GET)) ? $_GET['wreply'] : ''
```

Comment:
This is the main problem I see, as it introduces an [_open redirection_ vulnerability](https://www.owasp.org/index.php/Top_10_2013-A10-Unvalidated_Redirects_and_Forwards).

Any URL there could be used by an attacker to force the response from the IdP to be sent to another place, under his/her control. We should therefore restrict the URLs allowed as return URLs, the sooner the better.

This could be fixed by adding the `adfs:wreply` key conditionally, only if the `wreply` query parameter is present, and the URL contained there is a valid URL and matches some whitelisted host, by calling `SimpleSAML\Utils\HTTP::isURLAllowed()` on it.